cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -113,8 +113,19 @@ public void configure(Map<String, ?> configs) {
 
     //Add BootstrapServers if not set
     if (!producerProps.containsKey(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)) {
-      Object port = configs.get("port");
-      String bootstrapServers = "localhost:" + (port == null ? "9092" : String.valueOf(port));
+      String listeners = configs.get("listeners") == null ? null : String.valueOf(configs.get("listeners"));
+      String bootstrapServers = null;
+      if (listeners != null && listeners.length() != 0) {
+        String firstListener = listeners.split("\\s*,\\s*")[0];
+        String[] allListeners = firstListener.split(":");
+        if (allListeners[1].length() == 2) {
```

Comment:
What is the expected structure if `allListeners[1].length() == 2` is `true` (and what is expected when it is `false`)?

Reply:
> What is the expected structure if `allListeners[1].length() == 2` is `true` (and what is expected when it is `false`)?
Thanks for your answer.
eg:
PLAINTEXT://:9092  'allListeners[1].length() == 2' is 'true' 
PLAINTEXT://myhost:9092、PLAINTEXT://localhost:9092   'allListeners[1].length() == 2' is 'false' 
but I found that if it is ipv6 it will not be established,let me modify